x/pairing/keeper/unstaking.go

```diff
@@ -62,51 +62,61 @@ func (k Keeper) UnstakeEntry(ctx sdk.Context, chainID string, creator string, un
 	return k.epochStorageKeeper.AppendUnstakeEntry(ctx, existingEntry, unstakeHoldBlocks)
 }
 
-func (k Keeper) CheckUnstakingForCommit(ctx sdk.Context) error {
+func (k Keeper) CheckUnstakingForCommit(ctx sdk.Context) {
 	// this pops all the entries that had their deadline pass
 	unstakingEntriesToCredit := k.epochStorageKeeper.PopUnstakeEntries(ctx, uint64(ctx.BlockHeight()))
 
 	if unstakingEntriesToCredit != nil {
-		err := k.creditUnstakingEntries(ctx, unstakingEntriesToCredit) // true for providers
-		if err != nil {
-			panic(err.Error())
-		}
+		k.creditUnstakingEntries(ctx, unstakingEntriesToCredit) // true for providers
 	}
+}
 
+func (k Keeper) refundUnstakingProvider(ctx sdk.Context, addr sdk.AccAddress, neededAmount sdk.Coin) error {
+	moduleBalance := k.bankKeeper.GetBalance(ctx, k.accountKeeper.GetModuleAddress(types.ModuleName), epochstoragetypes.TokenDenom)
+	if moduleBalance.IsLT(neededAmount) {
+		return fmt.Errorf("insufficient balance to unstake %s (current balance: %s)", neededAmount, moduleBalance)
+	}
+	err := k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, addr, []sdk.Coin{neededAmount})
+	if err != nil {
+		return fmt.Errorf("failed to send coins from module to %s: %w", addr, err)
+	}
 	return nil
 }
 
 func (k Keeper) creditUnstakingEntries(ctx sdk.Context, entriesToUnstake []epochstoragetypes.StakeEntry) error {
```

Comment:
if we always return nil this func should not have a return value

Reply:
good catch; forgot to change this.